flash/video/classification/data.py

```diff
@@ -174,6 +363,121 @@ def from_data_frame(
+            predict_input_cls: The :class:`~flash.core.data.io.input.Input` type to use for loading the prediction data.
+            transform_kwargs: Dict of keyword arguments to be provided when instantiating the transforms.
+            data_module_kwargs: Additional keyword arguments to provide to the
+                :class:`~flash.core.data.data_module.DataModule` constructor.
+
+        Returns:
+            The constructed :class:`~flash.video.classification.data.VideoClassificationData`.
+
+        Examples
+        ________
+
+        .. testsetup::
+
+            >>> import os
+            >>> import torch
+            >>> from torchvision import io
+            >>> data = torch.randint(255, (10, 64, 64, 3))
+            >>> os.makedirs("train_folder", exist_ok=True)
+            >>> os.makedirs("predict_folder", exist_ok=True)
+            >>> io.write_video(os.path.join("train_folder", "video_1.mp4"), data, 5, "libx264rgb", {"crf": "0"})
+            >>> io.write_video(os.path.join("train_folder", "video_2.mp4"), data, 5, "libx264rgb", {"crf": "0"})
+            >>> io.write_video(os.path.join("train_folder", "video_3.mp4"), data, 5, "libx264rgb", {"crf": "0"})
+            >>> _ = [
+            ...     io.write_video(
+            ...         os.path.join("predict_folder", f"predict_video_{i}.mp4"), data, 5, "libx264rgb", {"crf": "0"}
+            ...     ) for i in range(1, 4)
+            ... ]
+
+        .. doctest::
+
+            >>> from pandas import DataFrame
+            >>> from flash import Trainer
+            >>> from flash.video import VideoClassifier, VideoClassificationData
+            >>> train_data_frame = DataFrame.from_dict(
+            ...     {
+            ...         "videos": ["video_1.mp4", "video_2.mp4", "video_3.mp4"],
+            ...         "targets": ["cat", "dog", "cat"],
+            ...     }
+            ... )
+            >>> predict_data_frame = DataFrame.from_dict(
+            ...     {
+            ...         "videos": ["predict_video_1.mp4", "predict_video_2.mp4", "predict_video_3.mp4"],
+            ...     }
+            ... )
+            >>> datamodule = VideoClassificationData.from_data_frame(
+            ...     "videos",
+            ...     "targets",
+            ...     train_data_frame=train_data_frame,
+            ...     train_videos_root="train_folder",
+            ...     predict_data_frame=predict_data_frame,
+            ...     predict_videos_root="predict_folder",
+            ...     transform_kwargs=dict(image_size=(244, 244)),
+            ...     batch_size=2,
+            ... )
+            >>> datamodule.num_classes
+            2
+            >>> datamodule.labels
+            ['cat', 'dog']
+            >>> model = VideoClassifier(backbone="x3d_xs", num_classes=datamodule.num_classes)
+            >>> trainer = Trainer(fast_dev_run=True)
+            >>> trainer.fit(model, datamodule=datamodule)  # doctest: +ELLIPSIS +NORMALIZE_WHITESPACE
```

Comment:
```suggestion
            >>> trainer.fit(model, datamodule=datamodule)  # doctest: +ELLIPSIS
```
with Doctest-plus this `NORMALIZE_WHITESPACE` shall be default

Reply:
Yes but not within PyCharm doctest runner, so at present the tests will pass from PyCharm